### DISCUSSION TITLE
Visa Compliance

### DISCUSSION THREAD

**Discussion by JRosielewska** [braintree/braintree_android#1021] [General]
Hi,

According to the new [Visa compliance](https://developer.paypal.com/braintree/articles/risk-and-security/compliance/network-updates/2024/v-s24#what-are-the-updates-to-visa-secure-data-fields-mandate) an app needs to provide Common Device Identification Parameters (Device IP Address):

![image](https://github.com/braintree/braintree_android/assets/97162606/04a785b1-8dff-45eb-a5cc-8843ae62b11b)

Do you know if a device's data collected from the [DataCollector](https://braintree.github.io/braintree_android/BraintreeDataCollector/com.braintreepayments.api/-data-collector/collect-device-data.html?query=open%20fun%20collectDeviceData(context:%20Context,%20callback:%20DataCollectorCallback)) object provided by Braintree is sufficient enough for these Visa new requirements?

**Comment by sarahkoop** [comment]
Hi @JRosielewska thanks for your question - we are looking into this and will report back when we have more info!
